api_v2/src/notifications/strategies/overtime-notification.strategy.ts

```diff
@@ -554,23 +565,54 @@ export class OvertimeNotificationStrategy implements NotificationStrategy {
     };
   }
 
-  private buildProjectsForManager(
+  private async buildProjectsForManager(
     overtime: any,
     projectNames: string[],
-  ): Array<{ Id: number | null; Name: string; ProjectLeader: any | null }> {
-    if (Array.isArray(overtime.projectLeadApprovers)) {
-      return overtime.projectLeadApprovers.map((mp: any) => ({
-        Id: mp.project?.id ?? null,
-        Name: mp.project?.name ?? '',
-        ProjectLeader: mp.project?.projectLeader
-          ? {
-              Id: mp.project.projectLeader.id,
-              Name: mp.project.projectLeader.name ?? '',
-              ApprovalStatus: RequestStatusEnum.PENDING,
-            }
-          : null,
-      }));
+  ): Promise<
+    Array<{ Id: number | null; Name: string; ProjectLeader: any | null }>
+  > {
+    if (
+      Array.isArray(overtime.projectLeadApprovers) &&
+      overtime.projectLeadApprovers.length > 0
+    ) {
+      const projects = await Promise.all(
+        overtime.projectLeadApprovers.map(async (mp: any) => {
+          const projectLeaderId = mp.projectLeaderId ?? null;
+
+          let projectLeader = null;
+          if (projectLeaderId) {
+            const projectLeaderData =
+              await this.userRepository.findUserById(projectLeaderId);
+            projectLeader = {
+              Id: projectLeaderId,
+              Name: projectLeaderData?.name ?? '',
+              ApprovalStatus:
```

Comment:
question: mp.status is same with `RequestStatusEnum.APPROVED` `RequestStatusEnum.DISAPPROVED` `RequestStatusEnum.PENDING` ?
```suggestion
              ApprovalStatus: mp.status
```

Reply:
My bad, Will refactor it